Content.Shared/Clothing/Components/ChameleonClothingComponent.cs

```diff
@@ -38,21 +43,23 @@ public sealed partial class ChameleonClothingComponent : Component
 public sealed class ChameleonBoundUserInterfaceState : BoundUserInterfaceState
 {
     public readonly SlotFlags Slot;
-    public readonly string? SelectedId;
+    public readonly EntProtoId? SelectedId;
+    public readonly EntityWhitelist? Whitelist;
 
-    public ChameleonBoundUserInterfaceState(SlotFlags slot, string? selectedId)
+    public ChameleonBoundUserInterfaceState(SlotFlags slot, EntProtoId? selectedId, EntityWhitelist? whitelist)
```

Comment:
You can just network it in the component and not need it in the BUI state at all.